make_prg/utils/io_utils.py

```diff
@@ -0,0 +1,221 @@
+import gzip
+import fileinput
+import shutil
+import uuid
+from Bio import AlignIO
+import os
+from loguru import logger
+from make_prg.from_msa import MSA
+from typing import Dict, Optional, List, Tuple
+from collections import defaultdict
+from pathlib import Path
+from zipfile import ZipFile
+
+
+def load_alignment_file(msa_file: [str, Path], alignment_format: str) -> MSA:
+    msa_file = str(msa_file)
+    if msa_file.endswith(".gz"):
+        handle = gzip.open(msa_file, "rt")
+        alignment = AlignIO.read(handle, alignment_format)
+        handle.close()
+    else:
+        alignment = AlignIO.read(msa_file, alignment_format)
+    for record in alignment:
+        record.seq = record.seq.upper()
+    return alignment
+
+
+def concatenate_text_files(input_filepaths, output_filepath):
+    output_filepath_parent_dir = Path(output_filepath).parent
+    os.makedirs(output_filepath_parent_dir, exist_ok=True)
+
+    with open(output_filepath, "w") as fout, fileinput.input(input_filepaths) as fin:
+        for line in fin:
+            fout.write(line)
+
+
+# Note: not unit tested
+# From https://gist.github.com/jacobtomlinson/9031697
+def remove_empty_folders(path: str, remove_root: bool = True):
+    if not os.path.isdir(path):
+        return
+
+    # remove empty subfolders
+    files = os.listdir(path)
+    if len(files):
+        for f in files:
+            fullpath = os.path.join(path, f)
+            if os.path.isdir(fullpath):
+                remove_empty_folders(fullpath)
+
+    # if folder empty, delete it
+    files = os.listdir(path)
+    if len(files) == 0 and remove_root:
+        os.rmdir(path)
+
+
+# Note: not unit tested
+def output_files_already_exist(output_prefix: str):
+    return (
+        Path(output_prefix + ".prg.fa").exists()
+        or Path(output_prefix + ".update_DS.zip").exists()
+        or Path(output_prefix + ".prg.bin").exists()
+        or Path(output_prefix + ".prg.bin.zip").exists()
+        or Path(output_prefix + ".prg.gfa").exists()
+        or Path(output_prefix + ".prg.gfa.zip").exists()
+    )
+
+
+def create_temp_dir(output_prefix: str) -> Path:
```

Comment:
why use a custom function for this when https://docs.python.org/3.6/library/tempfile.html exists?

Reply:
You are right, now using `tempfile.mkdtemp()` (https://github.com/iqbal-lab-org/make_prg/pull/34/commits/ee0a383713c81630fc7505e6d901c07da83d76a6)